File: LLD-Python/solid_principle/liskov_substitution_principle/liskov_substitution_good.py
from abc import ABC, abstractmethod

class Bird(ABC):
    @abstractmethod
    def eat(self) -> None:
        ...
        
class FlyingBird(Bird):
    @abstractmethod
    def fly(self) -> None:
        ...

class Sparrow(FlyingBird):
    def fly(self)-> None:
        pass
    def eat(self)-> None:
        pass

File: LLD - Golang/SolidPrinciple/LiskovSubstitutionPrinciple/LiskovSubstitutionGood.go
package liskovsubstitutionprinciple

type BirdGood interface {
	Eat()
}

type FlyingBird interface {
	BirdGood
	Fly()
}

type Sparrow struct{}

func (Sparrow) Eat() {}
func (Sparrow) Fly() {}

type PenguinGood struct{}

func (PenguinGood) Eat() {}
